src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import React from 'react'
+import hashStr from '../vendor/glamor/hash'
+import StyleSheet from '../models/StyleSheet'
+import type { Interpolation, Stringifier } from '../types'
+
+const _createGlobalStyle = (stringifyRules: Stringifier, css: Function) => {
+  const createGlobalStyle = (strings: Array<string>, ...interpolations: Array<Interpolation>) => {
+    return class extends React.Component {
+      componentId = ''
+
+      componentDidMount() {
+        const rules = css(strings, ...interpolations)
+        const hash = hashStr(JSON.stringify(rules))
+        this.componentId = `sc-global-${hash}`
+        if (StyleSheet.instance.hasInjectedComponent(this.componentId)) return
```

Comment:
This needs to possible grab the stylesheet from the context (for SSR as well, and/or future tag support)

See here: https://github.com/styled-components/styled-components/blob/master/src/models/StyledComponent.js#L94

and here: https://github.com/styled-components/styled-components/blob/master/src/models/StyledComponent.js#L274-L277

Reply:
Sweet. Thanks for the tips. Adjusted in https://github.com/styled-components/styled-components/pull/1416/commits/c6155ec3680a5639673e24073f5e802a2ff2ca3c